### PR TITLE
fix pdm issue with #352

### PR DESCRIPTION
**Pull request by hathach**
- also add define for PIN_PDM_DIN, PIN_PDM_CLK, PIN_PDM_PWR
- add PDM setPins() for custom board

### CODE REVIEW

File: libraries/PDM/src/PDM.cpp [hunk 1]
```diff
@@ -203,13 +214,24 @@ void PDMClass::IrqHandler()
...
+
+#ifndef PIN_PDM_CLK
+#define PIN_PDM_CLK -1
+#endif
+
+#ifndef PIN_PDM_PWR
+#define PIN_PDM_PWR -1
+#endif
+
+PDMClass PDM(PIN_PDM_DIN, PIN_PDM_CLK, PIN_PDM_PWR);
```
Comment (hathach) [inline libraries/PDM/src/PDM.cpp]: PDM is declared within the class like Serial or Wire. It is consistent with current usage of PDM from Arduino nRF528x-mbedos. Making using 3rd party library easier. 

Note: it requires to remove this line from CPlay library https://github.com/adafruit/Adafruit_CircuitPlayground/blob/master/utility/Adafruit_CPlay_Mic.cpp#L30 . I will submit an PR for this after this once is merged.

File: libraries/PDM/src/PDM.h [hunk 2]
```diff
@@ -33,6 +33,7 @@ class PDMClass
   PDMClass(int dataPin, int clkPin, int pwrPin);
   virtual ~PDMClass();
 
+  void setPins(int dataPin, int clkPin, int pwrPin);
```
Comment (hathach) [inline libraries/PDM/src/PDM.h]: add setPins(), will be useful for custom nrf52 board.

File: variants/circuitplayground_nrf52840/variant.h [hunk 3]
```diff
@@ -134,6 +136,13 @@ static const uint8_t SCK  = PIN_SPI_SCK ;
 #define USB_MSC_BLOCK_SIZE    512
 #define USB_MSC_BLOCK_COUNT   ((2*1024*1024) / USB_MSC_BLOCK_SIZE)
 
+/*
+ * PDM Interfaces
+ */
+#define PIN_PDM_DIN         24
```
Comment (hathach) [inline variants/circuitplayground_nrf52840/variant.h]: currently these pin are defined for CPB

File: libraries/PDM/examples/PDMSerialPlotter/PDMSerialPlotter.ino [hunk 4]
```diff
@@ -10,10 +10,6 @@
 
 #include <PDM.h>
 
-// The default Circuit Playground Bluefruit pins 
-// data pin, clock pin, power pin (-1 if not used)
-PDMClass PDM(24, 25, -1);
```
Comment (hathach) [inline libraries/PDM/examples/PDMSerialPlotter/PDMSerialPlotter.ino]: removed, we could use PDM directly like we did with Serial

File: libraries/PDM/src/PDM.cpp [hunk 5]
```diff
@@ -145,7 +158,7 @@ int PDMClass::available()
 
   size_t avail = _doubleBuffer.available();
 
-  //NVIC_EnableIRQ(PDM_IRQn);
+  NVIC_EnableIRQ(PDM_IRQn);
```
Comment (henrygab) [inline libraries/PDM/src/PDM.cpp]: By disabling the PDM in the NVIC, wouldn't this potentially cause the silent loss of samples of data?
Especially with softdevice enabled, and Bluetooth active, which can pre-empt the execution here and cause unexpectedly long delays while this is still disabled?

File: libraries/PDM/src/PDM.cpp [hunk 6]
```diff
@@ -116,7 +128,6 @@ int PDMClass::begin(int channels, long sampleRate)
   nrf_pdm_event_clear(NRF_PDM_EVENT_STARTED);
   nrf_pdm_task_trigger(NRF_PDM_TASK_START);
```
Comment (henrygab) [inline libraries/PDM/src/PDM.cpp]: The register buffer pointer is still set to `nullptr`.  Can you help me understand why this might be intentional?

**Comment by hathach** [comment]
> I think this may not result in a working PDM library.
> 
> It appears that buffers are provided (via the callback), before they are actually filled by the DMA engine.
> 
> I'd like to better understand how using the NVIC disable / enable for the public APIs will (or will not) risk losing samples.

@henrygab thank you very much for analyzing the PDM implementation, unfortunately I am not capable of answering your question since I just learned about PDM recently. I mostly ste..l hmm use the code from Arduino repo + code comparing. You should really file issue there to get answer from PDM writer.  
https://github.com/arduino/ArduinoCore-nRF528x-mbedos/blob/master/libraries/PDM/src/PDM.cpp

**Pull request merged**